src/Model/Model.ts

```diff
@@ -489,16 +489,33 @@ export class Model<T extends IModelData> implements IModel<T> {
     const service = getDataService(this.serviceName);
     observable
       .takeUntil(this.getWillDestroyObservable$())
-      .subscribe((value) => {
+      .subscribe((updatedValue => {
         if (!this.relatedModels.hasOwnProperty(fieldName)) {
-          this.relatedModels[fieldName] = value;
-        } else if (this.relatedModels[fieldName] !== value) {
-          service
-            .actions
-            .setRelationship({ id: this.id, fieldName, value })
-            .invoke();
+          this.relatedModels[fieldName] = updatedValue;
+        } else if (this.relatedModels[fieldName] !== updatedValue) {
+          const currentValue: Model<any> | Model<any>[] | any = this.relatedModels[fieldName];
+
+          if (currentValue instanceof Array) {
+            currentValue.forEach(related => {
+              if (!related.isDestroying) {
+                // destroy model, so that the subscription above is canceled
+                related.markForDestruction();
+              }
+            });
+            service
+              .actions
+              .setRelationship({ id: this.id, fieldName, value: updatedValue })
+              .invoke();
```

Comment:
for whatever reason, we always need to call setRelationship even when the related isDestroying, it doesn't have the same issues as the unary operation here.